llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -245,6 +245,16 @@ AspectsSetTy getAspectsUsedByInstruction(const Instruction &I,
     Result.insert(Aspects.begin(), Aspects.end());
   }
 
+  // Opaque pointer arguments may hide types of pointer arguments until elements
+  // inside the types are accessed through a GEP instruction. However, this will
+  // not be caught by the operands check above, so we must extract the
+  // information directly from the GEP.
```

Comment:
I'd prefer to see a minimal llvm IR `opt`-based test under `llvm/test` as well.

Reply:
I have added llvm/test/SYCLLowerIR/PropagateAspectsUsage/gep-prop.ll. Is that what you had in mind?